pkg/engines/tekton/controller.go

```diff
@@ -1,743 +1,127 @@
-		// more up to date that when the item was initially put onto the
-		// workqueue.
-		if key, ok = obj.(string); !ok {
-			// As the item in the workqueue is actually invalid, we call
-			// Forget here else we'd go into a loop of attempting to
-			// process a work item that is invalid.
-			c.queue.Forget(obj)
-			c.logger.Warnf("expected string in workqueue but got %#v", obj)
+// SetupWithManager sets up the reconcilier with it's manager
+func (r *LighthouseJobReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	if err := mgr.GetFieldIndexer().IndexField(&pipelinev1beta1.TaskRun{}, jobOwnerKey, func(rawObj runtime.Object) []string {
+		obj := rawObj.(*pipelinev1beta1.TaskRun)
+		owner := metav1.GetControllerOf(obj)
+		if owner == nil {
 			return nil
 		}
-		// Run the syncHandler, passing it the namespace/name string of the
-		// PipelineActivity resource to be synced.
-		if err := c.syncHandler(key); err != nil {
-			// Put the item back on the workqueue to handle any transient errors.
-			c.queue.AddRateLimited(key)
-			return fmt.Errorf("error syncing '%s': %s, requeuing", key, err.Error())
-		}
-		// Finally, if no error occurs we Forget this item so it does not
-		// get queued again until another change happens.
-		c.queue.Forget(obj)
-		c.logger.Infof("Successfully synced '%s'", key)
-		return nil
-	}(obj)
-
-	if err != nil {
-		c.logger.WithError(err).Error("failure reconciling")
-		return true
-	}
-
-	return true
-}
-
-// syncHandler compares the actual state with the desired, and attempts to
-// converge the two.
-func (c *Controller) syncHandler(key string) error {
-	// Get the type of the key
-	keyParts := strings.Split(key, ":::")
-
-	if len(keyParts) != 2 {
-		return fmt.Errorf("no key type found in %s", key)
-	}
-	// Convert the namespace/name string into a distinct namespace and name
-	namespace, name, err := cache.SplitMetaNamespaceKey(keyParts[1])
-	if err != nil {
-		c.logger.Warnf("invalid resource key: %s", keyParts[1])
-		return nil
-	}
-
-	if keyParts[0] == prKeyPrefix {
-		return c.syncPipelineRun(namespace, name, key)
-	}
-	if keyParts[0] == jobKeyPrefix {
-		return c.syncJob(namespace, name, key)
-	}
-
-	return fmt.Errorf("unknown key type: %s", keyParts[0])
-}
-
-func (c *Controller) syncJob(namespace, name, key string) error {
-	// Get the LighthouseJob resource with this namespace/name
-	origJob, err := c.lhLister.LighthouseJobs(namespace).Get(name)
-	if err != nil {
-		if kubeerrors.IsNotFound(err) {
-			c.logger.Warnf("job '%s' in work queue no longer exists", key)
+		if owner.APIVersion != apiGVStr || owner.Kind != "Project" {
 			return nil
 		}
-
-		// Return an error here so that we requeue and retry.
+		return []string{owner.Name}
+	}); err != nil {
 		return err
 	}
 
-	spec := &origJob.Spec
-
-	// Only launch for the appropriate agent types and for triggered state
-	if origJob.Status.State == v1alpha1.TriggeredState && spec.Agent == config.TektonPipelineAgent {
```

Comment:
@abayer jobs use only `TriggeredState` and `PendingState` ?
Looks like the state property is not updated to other state.

Reply:
https://github.com/jenkins-x/lighthouse/blob/master/pkg/foghorn/controller.go#L265 - we update it in `foghorn` from the activity record on the status.

Reply:
👍

Reply:
My reasoning for doing this in `foghorn` is to have the engine controllers (Tekton and https://github.com/jenkins-x/lighthouse-jx-controller/) do as little as possible, so their job is to handle triggering and to update the status fields that require engine-specific information/decisions (i.e., `Activity` and `ReportURL`), while `foghorn` waits for the engine controllers to make changes to those fields and then does "real" updates to the status and to the SCM provider, etc.

It may not be a great architecture, so I'm not unopposed to changing it, but it feels like it makes sense to me.

Reply:
Yeah, i wondered what was the purpose of `ActivityRecord`. Now i know !
It probably makes sense but it's not easy to get it at first sight.